tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -1152,6 +1152,40 @@ public function testModelAttributesAreCastedWhenPresentInCastsArray()
 	}
 
 
+	public function testModelAttributeCastingPreservesNull()
```

Comment:
The way I see it, I'm testing one single thing: **no casting should be done for `null`**. Period.